django/contrib/gis/db/backends/postgis/schema.py

```diff
@@ -82,3 +60,31 @@ def _alter_column_type_sql(
 
     def _create_spatial_index_name(self, model, field):
         return self._create_index_name(model._meta.db_table, [field.column], "_id")
+
+    def _create_spatial_index_sql(self, model, field, **kwargs):
+        expressions = None
+        opclasses = None
+        fields = [field]
+        if field.geom_type == "RASTER":
+            # For raster fields, wrap index creation SQL statement with ST_ConvexHull.
+            # Indexes on raster columns are based on the convex hull of the raster.
+            expressions = Func(Col(None, field), template=self.rast_index_template)
+            fields = None
+        elif field.dim > 2 and not field.geography:
+            # Use "nd" ops which are fast on multidimensional cases
+            opclasses = [self.geom_index_ops_nd]
+        if not (name := kwargs.get("name")):
+            name = self._create_spatial_index_name(model, field)
+
+        return super()._create_index_sql(
+            model,
+            fields=fields,
+            name=name,
+            using=" USING %s" % self.geom_index_type,
+            opclasses=opclasses,
+            expressions=expressions,
+        )
+
+    def _delete_spatial_index_sql(self, model, field):
+        index_name = self._create_spatial_index_name(model, field)
+        return self._delete_index_sql(model, index_name)
```

Comment:
I think this is unused at this point (i.e by the second commit) but gets used in the next/third commit: c4dfc842435ed2eee47d6f6d5ed3f1f6b3bb3771
How do you feel if I move this `_delete_spatial_index_sql` into the third commit? 

It's less symmetrical than being added here, but I'm hoping to merge in the first two commits and this makes it feel slightly more coupled to the third commit

Reply:
Both works for me :+1: